### PR TITLE
Updating admin_ack() for multiple gates in the same time

### PR DESCRIPTION
**Pull request by JianLi-RH**

### CODE REVIEW

**Comment by JianLi-RH** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-nightly-4.14-upgrade-from-stable-4.14-vsphere-upi-f28

**Comment by JianLi-RH** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-4.15-upgrade-from-stable-4.14-azure-ipi-proxy-workers-rhcos-rhel8-f28

**Comment by JianLi-RH** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-nightly-4.14-upgrade-from-stable-4.7-gcp-ipi-f30

**Comment by jianlinliu** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-stable-4.14-upgrade-from-stable-4.13-gcp-ipi-proxy-private-f14

**Comment by jianlinliu** [comment]
From the [log](https://gcsweb-qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/qe-private-deck/pr-logs/pull/openshift_release/44865/rehearse-44865-periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-stable-4.14-upgrade-from-stable-4.13-gcp-ipi-proxy-private-f14/1717724851327733760/artifacts/gcp-ipi-proxy-private-f14/cucushift-upgrade-toimage/build-log.txt), the change works, does not break anyting.

**Comment by JianLi-RH** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-nightly-4.14-upgrade-from-stable-4.7-gcp-ipi-f30

**Comment by JianLi-RH** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-nightly-4.14-upgrade-from-stable-4.7-gcp-ipi-f30

**Comment by JianLi-RH** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-arm64-nightly-4.13-upgrade-from-stable-4.12-azure-ipi-ingress-custom-controller-f28

**Comment by JianLi-RH** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-4.15-upgrade-from-stable-4.14-aws-ipi-shared-vpc-phz-f14

**Comment by JianLi-RH** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.12-amd64-stable-4.12-upgrade-from-eus-4.10-gcp-ipi-ovn-ipsec-f14

**Comment by JianLi-RH** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-arm64-nightly-4.14-upgrade-from-eus-4.12-azure-ipi-f28

**Comment by JianLi-RH** [comment]
hi @jiajliu @shellyyang1989 @jianlinliu , it seems admin-ack patched successfully for both eus and normal upgrade, shell we approve and merge this PR?

https://storage.googleapis.com/qe-private-deck/pr-logs/pull/openshift_release/44865/rehearse-44865-periodic-ci-openshift-openshift-tests-private-release-4.12-amd64-stable-4.12-upgrade-from-eus-4.10-gcp-ipi-ovn-ipsec-f14/1718481072695545856/build-log.txt?Expires=1698628113&GoogleAccessId=qe-private-deck%40openshift-ci-private.iam.gserviceaccount.com&Signature=Yv5EwlRenrC0tMIACvJ%2B%2F1njZbZUpAVyv%2BouovZ1FajJF9hQ58T1bsHnPSVA%2BSYZErHvh%2Fv0ImzVjxafGTSBUEwGLVV2ZKnWL7p6xuQd%2BunFopW17S1naEIYZXpeiUonc2T%2Fechq4iTnSR0utWN3N%2BRCBJDsV13QtotBOcsW3P2%2FIdSI8FnAOJ1hPqSFJYZE4P2PGdxisVw0PzVhTtNQha48PpTLHziNdJBc51qpkGpAIhCdItgFcfBbn8vE9STK6ip%2F28pHJS8G7zqtmI8eE4fYdIAV5KJdRylg0raPtwBYVvdx94ONIXt37oAzdELcMvskcrpnqS6b2uiFo0DTbA%3D%3D

File: ci-operator/step-registry/cucushift/upgrade/paused/intermediate/to-latest/cucushift-upgrade-paused-intermediate-to-latest-commands.sh [hunk 1]
```diff
@@ -82,6 +90,9 @@ function check_upgrade_status() {
         if [[ ${avail} == "True" && ${progress} == "False" && ${out} == *"Cluster version is ${INTERMEDIATE_Y_VERSION}"* ]]; then
             echo -e "Upgrade succeed on $(date "+%F %T")\n\n"
             return 0
+        else
```
Comment (jianlinliu) [inline ci-operator/step-registry/cucushift/upgrade/paused/intermediate/to-latest/cucushift-upgrade-paused-intermediate-to-latest-commands.sh:93]: I guess this is some unexpected change introduced from your rebase.

File: ci-operator/step-registry/cucushift/upgrade/paused/latest/to-latest/cucushift-upgrade-paused-latest-to-latest-commands.sh [hunk 2]
```diff
@@ -64,6 +72,9 @@ function check_upgrade_status() {
         if [[ ${avail} == "True" && ${progress} == "False" && ${out} == *"Cluster version is ${TARGET_Y_VERSION}"* ]]; then
             echo -e "Upgrade succeed on $(date "+%F %T")\n\n"
             return 0
+        else
+            sleep 5m
+            (( wait_upgrade -= 5 ))
```
Comment (jianlinliu) [inline ci-operator/step-registry/cucushift/upgrade/paused/latest/to-latest/cucushift-upgrade-paused-latest-to-latest-commands.sh:77]: unexpected change from your rebase?

Comment (JianLi-RH) [reply ci-operator/step-registry/cucushift/upgrade/paused/latest/to-latest/cucushift-upgrade-paused-latest-to-latest-commands.sh]: Yes, introduced by the rebase, removed.

**Pull request merged**